### DISCUSSION TITLE
useSearch Breaks Non-root Routes

### DISCUSSION THREAD

**Discussion by ohthepain** [TanStack/router#2446] [General]
We want to move our <LeftColumn> out of the Outlet. But when I did that I started getting 

> Invariant failed: Could not find an active match from "/"

for all routes except the root. I removed as much as I could from the <LeftColumn> component and it starts working again when I remove the call to useSearch. 

Here is our route setup:

```
const rootRoute = createRootRoute({
  component: () => (
    <div className="flex flex-col h-screen root">
      <Header />
      <div className="flex flex-row w-screen">
        <LeftColumn />
        <Outlet />
      </div>
      <TanStackRouterDevtools />
    </div>
  ),
});

const indexRoute = createRoute({
  getParentRoute: () => rootRoute,
  path: '/',
  component: App,
  loader: loadRouteData,
  notFoundComponent: notFoundComponent,
});

const bomRoute = createRoute({
  getParentRoute: () => rootRoute,
  path: '/bom',
  component: BomPage,
  loader: loadRouteData,
  notFoundComponent: notFoundComponent,
});

export const routeTree = rootRoute.addChildren([indexRoute, bomRoute]);

```

Is this expected, a bug in the router, or am I doing something wrong?

**Comment by schiller-manuel** [comment]
Can you please provide a minimal complete example, e.g. by forking one of the examples on stackblitz?
